src/logging.cc

```diff
@@ -52,7 +52,14 @@ bool ConsoleLogger::ShouldLog(LogVerbosity verbosity) {
 }
 
 void ConsoleLogger::Configure(Args const& args) {
-  param_.UpdateAllowUnknown(args);
+  auto unknown = param_.UpdateAllowUnknown(args);
+  if (!unknown.empty()) {
+    std::ostringstream oss;
+    for (const auto& e : unknown) {
+      oss << e.first << ", ";
+    }
+    LOG(FATAL) << "Unknown parameters: " << oss.str();
```

Comment:
Ah, you are already using it.  Can we place this outside of logger?

Reply:
Actually I will need to remove this check from `ConsoleLogger::Configure()`, as it's failing lots of tests.